### DISCUSSION TITLE
Running Wildfly 15 Remote, Junit 5, Arquillian 1.7,  Wildfly-Arquillian 4 latest, Java 8 - Can't deploy when HTTPS is enabled in management console using self-signed certificate

### DISCUSSION THREAD

**Discussion by junrrein** [wildfly/wildfly-arquillian#292] [Q&A]
Hello there,  
When I try to deploy my tests under these conditions through the HTTP protocol with the following options in `arquillian.xml`:

```xml
<property name="managementAddress">...</property>
<property name="managementPort">9990</property>
<property name="username">...</property>
<property name="password">...</property>
```

I get [this error](https://gist.github.com/junrrein/c2c4c2cecfcc33082e5dabd79194011e).  
The relevant line appears to be:

```
Caused by: org.xnio.http.RedirectException: XNIO000816: Redirect encountered establishing connection
```

which I guess happens because, when trying to connect to the management console using HTTP, it answers back with a redirect asking to use HTTPS instead, which happens when using `jboss-cli`.  
(If my analisis is correct, I don't know why a redirect would be problem for the deployment process).

If I modify my `arquillian.xml` to deploy using HTTPS instead:

```xml
<property name="managementProtocol">remote+https</property>
<property name="managementAddress">mdcobro.local</property>
<property name="managementPort">9993</property>
<property name="username">admin</property>
<property name="password">admin</property>
```

I get [this error](https://gist.github.com/junrrein/9a8db39b9cde1215ec2704ca431e3008) instead.  
The relevant exception seems to be:

```
Caused by: sun.security.provider.certpath.SunCertPathBuilderException: unable to find valid certification path to requested target
```

which makes sense, since my Wildfly instance is using a self-signed certificate.

If I disable HTTPS for the management console, and I try to deploy using HTTP with the first config I showed in this discussion, then Arquillian can deploy my tests correctly.  

The thing is, I would rather not disable HTTPS because it's not desirable to create a difference between my development and production environments.  
Also, I would rather not import the self-signed certificate to my trust store since I would need to do that for every project where I use Arquillian for integration testing. Every project has its own VM replicating the production environment, and those VMs could be destroyed and recreated as a matter of routine, meaning I would have to import the certificates again, which would create friction when it comes to running these tests.

Disabling HTTPS is a simple enough solution, but is there another workaround for my situation?  
For example, I was thinking about configuring Arquillian to ignore TLS certificate validity when deploying. Can I do that?

I'm afraid I can't provide a minimally reproducing example, but I'll supply whatever debug info that's needed.

**Comment by jamezp** [comment]
I'm not sure if there is a way to override the TLS certificate. However, I will have a look at this as soon as I can. This seems like if it doesn't work, it's a reasonable feature request.

**Reply by junrrein** [reply]
Thank you for your consideration.

For reference, I have no problem deploying my application under the same circumstances using the [Gradle Wildfly Deploy plugin](https://plugins.gradle.org/plugin/com.mkring.wildlydeplyplugin.deploy-wildfly-plugin) through HTTP.  
That plugin can't deploy through HTTPS because it hardcodes the `remote+http` protocol in the connection's URL.
